R/powerEmp.R

```diff
@@ -73,42 +74,47 @@ powerEmp <- function(simTrials, criticalPFS, criticalOS, eventNumPFS, eventNumOS
   assert_list(simTrials, null.ok = FALSE)
   assert_positive_number(criticalPFS)
   assert_positive_number(criticalOS)
-  assert_int(eventNumPFS, lower = 1L)
-  assert_int(eventNumOS, lower = 1L)
+  assert_count(eventNumPFS, positive = TRUE)
+  assert_count(eventNumOS, positive = TRUE)
 
   nRep <- length(simTrials)
   simTrials <- lapply(simTrials, function(x) if (ncol(x) == 9) getDatasetWideFormat(x) else x)
 
-  # Censor simulated trials at time-point of OS/PFS analysis
-  trialsAnaPFS <- lapply(simTrials, censoringByNumberEvents,
-    eventNum = eventNumPFS, typeEvent = "PFS"
-  )
-  trialsAnaOS <- lapply(simTrials, censoringByNumberEvents,
-    eventNum = eventNumOS, typeEvent = "OS"
-  )
+  # Helper function to conduct log-rank tests for either PFS or OS:
+  passedLogRank <- function(typeEvent, eventNum, critical) {
```

Comment:
this could be good, but please move it out of the `powerEmp` function - because only then you can document and test separately

Reply:
thx @danielinteractive. Can I infer from this that we want "any" function defined separately if it is not just a one-liner or a very simple if-else structure (or sth. similar)?